rules.go

```diff
@@ -225,15 +232,38 @@ func gocriticArgOrder(m fluent.Matcher) {
 		Suggest(`strings.Contains($s2, $s1)`)
 }
 
+func badSplitNValue(m fluent.Matcher) {
+	// Interpret any non-positive match count as "match all" and replace it by a better function.
+	// The special case is zero. Zero does nothing, so it's a bug, so we replace it too.
+	// Everything else is for consistency.
+	m.Match(`strings.Replace($s, $d, $w, $n)`).
+		Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`)).
```

Comment:
It looks like if we had a way to const-fold expressions into int we could just check it like:
```
m["n"].ConstValue <= 0
```
It's not hard to implement, actually.

Reply:
```diff
- Where(m["n"].Const && m["n"].Text.Matches(`(\-[0-9]+|0)`))
+ Where(m["n"].Value.Int() <= 0)
```